types/index.d.ts

```diff
@@ -205,7 +205,7 @@ declare module 'mongoose' {
    * http://mongoosejs.com/docs/api.html#drivers-node-mongodb-native-collection-js
    */
   let Collection: Collection;
-  interface Collection<T = AnyObject> extends CollectionBase<T> {
+  interface Collection<T extends AnyObject = AnyObject> extends CollectionBase<T> {
```

Comment:
I used `AnyObject` here for readability, but `mongodb.Document` and `AnyObject` are interchangeable because they have the same definition.

Reply:
maybe we should use mongodb.Document, because what happens if they change their typing?

Reply:
That makes more sense, especially since CollectionBase has the same constraint on its type parameter.